swift.en.md

```diff
@@ -0,0 +1,340 @@
+
+```swift
+// Bad
+var foodCount: Int {
+    get { return foods.count }
+}
+
+// Good
+var foodCount: Int { 
+    return foods.count 
+}
+```
+
+- **[SHOULD]** If setting a property has a side-effect, write it in `willSet` or `didSet`.
+
+### Variable
+
+- **[MUST]** Do not use global variables.
+
+
+## Type
+
+- **[MUST]** Do not use `Array<T>`, `Dictionary<K, V>`. Use syntax sugar instead.
+
+```swift
+// Bad
+let seasonings: Array<String> = ["sugar", "salt", "vinegar", "soy sauce", "miso"]
+let ingredients: Dictionary<String, String> = [
+    "bacon": "100g",
+    "lettuce": "1/4",
+    "tomato": "half",
+]
+
+// Good let seasonings: [String] = ["sugar", "salt", "vinegar", "soy sauce", "miso"] let ingredients: [String: String] = [ "bacon": "100g",
+    "lettuce": "1/4",
+    "tomato": "half",
+]
+```
+
+- **[SHOULD]** Use `typealias` to represent complex type definitions.
+
+```swift
+// Good
+typealias RecipeClientCompletionBlock = (Result<[Recipe], APIError>) -> Void
+```
+
+- **[SHOULD]** Use classes without `NS` prefixes as possible.
+
+```swift
+// Bad
+let url = NSURL(string: "https://cookpad.com/")
+
+// Good
+let url = URL(string: "https://cookpad.com/")
+```
+
+
+- **[SHOULD]** Do not use `UInt`
+    - [Apple announced to avoid using it.](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/TheBasics.html#//apple_ref/doc/uid/TP40014097-CH5-ID320)
+
+## Enum
+
+- **[MUST]** Use lowerCamelCase as value naming.
+- **[MUST]** Drop enum type if you can.
+
+```swift
+// Bad
+user.status = UserStatus.guest
+
+// Good
+user.status = .guest
+```
+
+## Optional
+
+- **[MUST]** Do not use `Optional<T>`. Use syntax sugar instead.
+- **[MUST]** Use non-Optional value as possible.
+- **[MUST]** Do not use `guard` without optional-binding.
+
+```swift
+// Bad
+guard isLoggedIn else {
+    return
+}
+
+// Good
+if !isLoggedIn {
+    return
+}
+```
+
+```swift
+// Good
+guard let user = manager.currentUser {
+    return
+}
+label.text = user.name
+```
+
+## Exception
+
+- **[MUST]** Do not use `NSException`.
+- **[SHOULD]** Do not use `try!`. Use `do` ~ `catch` instead.
+
+## Naming
+
+- **[MUST]** Follow [Swift API Design Guideline](https://swift.org/documentation/api-design-guidelines/)
+- **[MUST]** Use `_` for unused returning values or arguments.
+- **[MUST]** Do not use vendor prefix.
+- **[MUST]** Follow same naming rule for constants as variables.
+- **[MUST]** To define constants, follow naming rule of variables.
+    - Do not use UpperCamelCase, SNAKE_CASE or `k` prefix as constants names.
+- **[MUST]** Do not use non-Ascii characters as variable names.
+- **[SHOULD]** Use nest structure to reperesent hierarchy.
+
+```swift
+// Bad
+enum RecipeType {
+    case none
+}
+
+class Recipe {
+}
+
+// Good
+class Recipe {
+    enum Type {
+        case none
+    }
+}
+
+```
+
+
+## Syntax
+
+- **[MUST]** Do not use `self` as possible. However, you can use it like following situations.
```

Comment:
**[MUST]** Do not use `self` when possible. However, you can use it in the following situations: